api/v1_coin.go

```diff
@@ -45,3 +39,25 @@ func (app *ApiServer) v1Coin(c *fiber.Ctx) error {
 		"data": coinRow,
 	})
 }
+
+func (app *ApiServer) v1Coin(c *fiber.Ctx) error {
```

Comment:
I'll approve it, but this is a bad code smell I think. Our pattern has been to establish as little indirection here as possible, i would prefer you copy the impl of getCoinByField into each of these and not do the string concat